### PR TITLE
Account for SequenceExpression in the utils

### PR DESCRIPTION
**Pull request by rrhg**
Fix for #95.
The prop value with an expression type of SequenceExpression could not be resolved.

### CODE REVIEW

File: src/values/expressions/SequenceExpression.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * Extractor function for a SequenceExpression type value node.
+ * A Sequence expression is an object with an attribute named 
+ * expressions which contains an array of different types 
+ *  of expression objects.
+ *
+ * @returns - An array of the extracted elements.
+ */
+export default function extractValueFromSequenceExpression(value) {
+  // eslint-disable-next-line global-require
+  const getValue = require('./index.js').default;
```
Comment (ljharb) [inline src/values/expressions/SequenceExpression.js:11]:
```suggestion
  const getValue = require('./').default;
```
(i realize there's other files that include the redundant "index.js" but let's not continue the mistake)

**Comment by rrhg** [comment]
Thanks !!! 
Could you provide some guidance. I'm not sure because I can't find how other "expression extractor functions" in [src/values/expressions/](https://github.com/jsx-eslint/jsx-ast-utils/tree/master/src/values/expressions) are been tested.

I don't have much experience on this :) . What comes to mind is creating a new test file just for testing `extractValueFromSequenceExpression()` & pass a SequenceExpression object.   

For example, playing with [https://github.com/reactjs/reactjs.org/issues/3120](https://github.com/reactjs/reactjs.org/issues/3120), I found out that, when running  `yarn lint` in the reactjs.org repo, the object been passed to  `extractValueFromSequenceExpression(object)` is :   

```
{"type":"SequenceExpression","start":821,"end":842,"loc":{"start":{"line":32,"column":12},"end":{"line":32,"column":33}},"range":[821,842],"expressions":[{"type":"Literal","start":821,"end":827,"loc":{"start":{"line":32,"column":12},"end":{"line":32,"column":18}},"range":[821,827],"value":"alt-","raw":"'alt-'","_babelType":"Literal"},{"type":"MemberExpression","start":829,"end":842,"loc":{"start":{"line":32,"column":20},"end":{"line":32,"column":33}},"range":[829,842],"object":{"type":"Identifier","start":829,"end":837,"loc":{"start":{"line":32,"column":20},"end":{"line":32,"column":28},"identifierName":"language"},"range":[829,837],"name":"language","_babelType":"Identifier"},"property":{"type":"Identifier","start":838,"end":842,"loc":{"start":{"line":32,"column":29},"end":{"line":32,"column":33},"identifierName":"code"},"range":[838,842],"name":"code","_babelType":"Identifier"},"computed":false,"_babelType":"MemberExpression"}],"extra":{"parenthesized":true,"parenStart":820},"_babelType":"SequenceExpression"}
```

**Comment by ljharb** [comment]
Here's a test for OptionalCallExpression: https://github.com/jsx-eslint/jsx-ast-utils/blob/f4752e6fdd0f8e6d7425a5c56e0627522dcf3ab4/__tests__/src/getPropValue-babelparser-test.js#L506-L522

**Comment by rrhg** [comment]
The test is failing & I can't figure out why. But it turns out the test you used as example above, was also already failing with this msg:

```
  ● getPropValue › Member expression › should evaluate to a correct representation of optional call expression

    assert.equal(received, expected) or assert(received) 
    
    Expected value to be (operator: ==):
      null
    Received:
      "bar.baz?.(quux)"
```

To make sure it was already failing before I touched it,  I cloned your repo again & run `yarn` & `yarn test`, and sure enough,  it is failing. So I'm confused about what to do next.

**Comment by rrhg** [comment]
I deleted previous test & created a new one in  __tests__/src/getPropValue-flowparser-test.js and now it passed.

The previous test was failing because of the babel parser. So I think now is in the right place.

Keep in mind that other 2 tests are failing, but those were failing before I touched the project.

**Comment by rrhg** [comment]
My apologies. Didn't realize had to do this small suggested change to the import. 

Please let me know if there's anything else I should do. Thanks

File: src/values/expressions/SequenceExpression.js (hunk 1 above)
Comment (rrhg) [reply src/values/expressions/SequenceExpression.js]: Yarn test gave me this error:
"  11:28  error  Useless path segments for "./", should be "."  import/no-useless-path-segments"

So ended up with: const getValue = require('.').default;

**Comment by ljharb** [comment]
looks like tests are failing :-/

**Comment by rrhg** [comment]
Yes. As I explained in previous comment, those test were failing before I touched to repo.  Maybe there's something I'm missing.

> To make sure it was already failing before I touched it, I cloned your repo again & run yarn & yarn test, and sure enough, it is failing. So I'm confused about what to do next.   

The tests failing are for file `getPropValue-babelparser-test.js`.
I initially put my test there but then deleted it & now my test is in file `getPropValue-flowparser-test.js`.  This file is not failing the test.

**Comment by ljharb** [comment]
ahh ok gotcha, i'll see if i can fix that up.

**Comment by rrhg** [comment]
Do you think is a good idea to merge this one & then I can try to do some research about the other failing tests ?

I spent some time looking at the code & maybe found the problem with [this test](https://github.com/jsx-eslint/jsx-ast-utils/blob/c86cea11051cfa7ce3f7ce3776631df75a076ab5/__tests__/src/getPropValue-babelparser-test.js#L488). The name implies a test for an expression of type  `MemberExpression` with a nullable member, but is receiving type `ChainExpression`.  

In `src/values/expressions/index.js` the expression object becomes null because the type `ChainExpression` doesn't exist.  Maybe it could be fix by creating the `ChainExpression` account in utils & add it to all necessary functions. 

I don't know the difference between  `MemberExpression` & `ChainExpression`. My guest is that:  

```
MemberExpression: <div foo={bar.baz} />
ChainExpression: <div foo={bar?.baz} />
``` 

Or maybe previously both were MemberExpression, & now they are divided ?. But you if guide me in the right direction, I can do it. 
Thanks

**Comment by ljharb** [comment]
Filed #102 to fix master.

**Review by ljharb — APPROVED** [review]

**Pull request merged**